src/slowmokit/methods/neighbors/knn/knn.hpp

```diff
@@ -8,9 +8,38 @@
 
 #include "../../../models/model.hpp"
 
-template <typename T>
-class KNN : public Model<T>
-{
+template <class T>
+class KNN{
+private:
+    std::vector<std::vector<T>> xTrain;
+    std::vector<int> yTrain;
+    int k=1;
+    int classNums=1;
+    /**
+     * @brief Calculates distance euclidean/else
+     * 
+     * @param x1 first point
+     * @param x2 second point
+     * @return double value of distance
+     */
+    double distance(std::vector<T> x1,std::vector<T> x2);
+public:
+    /**
+     * @brief store values of training data
+     * 
+     * @param x all x training values
+     * @param y all y training values
+     * @param k minimum number of nearest neighbors
+     * @param classNums number of classes
+     */
+    void fit(std::vector<std::vector<T>> x,std::vector<int> y,int k,int classNums);
+    /**
+     * @brief predict to which cluster it belongs
```

Comment:
Use the term class instead of cluster as "cluster" is used in the case of Unsupervised Algorithms